### PR TITLE
InputNode readonly option

### PR DESCRIPTION
**Pull request by sunag**
Not use `uniform` input if `readonly` is true in `InputNode`. ( for optimization )

in `new THREE.Vector3Node( 1, 1, 1 )` instead use explicit declaration, for example:
```glsl
// readonly = true
vec3( 1.0, 1.0, 1.0 )

// readonly = false (default)
uniform vec3 nV0;
```

Of course. `readonly` not allow change the value after build the shader material.

---

In `NodeFrame` I rename `updateFrame` to `updateNode` which is more suggestive but keeping `updateFrame` in nodes.

### CODE REVIEW

**Comment by sunag** [comment]
@mrdoob @WestLangley What do you think about rename the property `coord` to `uv` of `TextureNode` and `CubeTextureNode`?

**Comment by sunag** [comment]
Other thing... 'transform' is a `position offset` and `normal` is a `normal offset`.

**Comment by sunag** [comment]
>> 'transform' is a position offset

my mistake. actually `transform` is a `local position`. 
to create an offset, is need:
```javascript
material.transform = new THREE.OperatorNode(
	new THREE.PositionNode(),
	displaceNode,
	THREE.OperatorNode.ADD
);
```

but `normal` is a `normal offset`. currently to create a `normalMap` is need:
```javascript
mtl.normal = new THREE.TextureNode( getTexture( "grassNormal" ) );
```

but it could be like this:
```javascript
mtl.normal = new THREE.NormalMapNode( new THREE.TextureNode( getTexture( "grassNormal" ) ) );
```

and allow `normal` slot to others customized `NormalNode`.

Are some thoughts...

**Comment by WestLangley** [comment]
`mtl.normal` should be renamed `mtl.normalMap`, but this looks like the right API to me:
```javascript
mtl.normalMap = new THREE.NormalMapNode( new THREE.TextureNode( getTexture( "grassNormal" ) ) );
```

**Comment by sunag** [comment]
however, `mtl.normal` is like as `mtl.normalOffset` in pratice

File: examples/js/nodes/inputs/FloatNode.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ Object.defineProperties( THREE.FloatNode.prototype, {
 	}
 } );
 
+THREE.FloatNode.prototype.generateReadonly = function ( builder, output, uuid, type, ns, needsUpdate ) {
+
+	var value = this.number;
+
+	return builder.format( ~~value !== value ? value : value + ".0", type, output );
```
Comment (mrdoob) [inline examples/js/nodes/inputs/FloatNode.js]: Minor thing: I don't think using these bitwise operations are worth it these days (speed vs readability).

Comment (sunag) [reply examples/js/nodes/inputs/FloatNode.js]: Instead what do you recommend `parseInt` or `Math.floor`?

Comment (sunag) [reply examples/js/nodes/inputs/FloatNode.js]: I think that `Math.floor` is "more" number-oriented.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**